repos/system_upgrade/common/actors/detectmissingnewlineingrubcfg/actor.py

```diff
@@ -0,0 +1,37 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.libraries.actor.detectmissingnewlineingrubcfg import detect_config_error
+from leapp.models import GrubConfigError
+from leapp.reporting import create_report, Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class DetectMissingNewLineInGrubCfg(Actor):
+    """
+    Check grub configuration for missing new line at its end.
+    """
+
+    name = 'detect_missing_new_line_in_grub_cfg'
+    consumes = ()
+    produces = (Report, GrubConfigError)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        config = '/etc/default/grub'
+        error_detected = detect_config_error(config)
+        if error_detected:
+            create_report([
+                reporting.Title('Detected a missing new line at the end of grub configuration file.'),
+                reporting.Summary(
+                    'The missing new line in /etc/default/grub causes booting issues when appending '
+                    'new entries to this file during the upgrade. '
+                    'Error is automatically fixed by add_upgrade_boot_entry actor.'
```

Comment:
Users don't care, nor now, what actor will do it. 
```suggestion
                    'Leapp will automatically fix this problem by appending the missing newline to the grub configuration file.'
```

Reply:
I've noticed that this message is copied from the other error detecting actor, however, my original points still stands. It is unlikely that users know how is leapp structured internally.